libcudacxx/include/cuda/std/__cccl/builtin.h

```diff
@@ -147,6 +147,32 @@
 #  define _CCCL_BUILTIN_EXPECT(...) __builtin_expect(__VA_ARGS__)
 #endif // _CCCL_CHECK_BUILTIN(builtin_expect)
 
+#if _CCCL_CHECK_BUILTIN(builtin_fmax) || defined(_CCCL_COMPILER_GCC)
+#  define _CCCL_BUILTIN_FMAXF(...) __builtin_fmaxf(__VA_ARGS__)
+#  define _CCCL_BUILTIN_FMAX(...)  __builtin_fmax(__VA_ARGS__)
+#  define _CCCL_BUILTIN_FMAXL(...) __builtin_fmaxl(__VA_ARGS__)
+#endif // _CCCL_CHECK_BUILTIN(builtin_fmax)
+
+// Below 11.7 nvcc treats the builtin as a host only function
+#if defined(_CCCL_CUDACC_BELOW_11_7)
+#  undef _CCCL_BUILTIN_FMAXF
+#  undef _CCCL_BUILTIN_FMAX
+#  undef _CCCL_BUILTIN_FMAXL
+#endif // _CCCL_CUDACC_BELOW_11_7
```

Comment:
Because we settle on the general design that we have one for enablement and one for workarounds of compiler bugs.

That makes it much easier to see where issues are and whether we can drop workarounds

Also the condition to enable is a chain of || but disablement is through && which requires braces

Reply:
> Because we settle on the general design 

It would be awesome to collect some of this design decisions in a libcu++ developer guide. I maintain some things here: https://github.com/NVIDIA/cccl/issues/2635#issuecomment-2441338539, but ultimately we should either have some real documentation or a wiki page.